### PR TITLE
API documentation

### PR DESCRIPTION
**Pull request by cantsin**
Addresses https://github.com/18F/ReVAL/issues/94

- Updates version to 0.7.0
- Bug fix: API should serialize instance, not just return validation data (otherwise PUT/PATCH methods become confusing)
- Bug fix: standardize on `stage` instead of `complete`
- Documentation:
  - Upload model statuses
  - Endpoints
  - Sample data
  - Examples for each API endpoint (in details/summary html blocks so it's not too overwhelming)
  - Example python script using `requests` that goes through the upload model lifecycle
  - [Documentation in md format (easier to read/review)](https://github.com/18F/ReVAL/blob/ad86df39594e501f81a865c3679d79ae82b5daa9/docs/api.md)

### CODE REVIEW

File: data_ingest/views.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def confirm_upload(request, upload_id):
     return render(request, "data_ingest/confirm-upload.html", data)
 
 
-def complete_upload(request, upload_id):
+def stage_upload(request, upload_id):
```
Comment (amymok) [inline data_ingest/views.py:153]: I thought we were going to use `complete` and not `stage`?

Comment (cantsin) [reply data_ingest/views.py:153]: I changed my mind, I hope that is OK :P If you prefer `complete` I can change, either way.

Comment (amymok) [reply data_ingest/views.py:153]: Okay :) I was just curious.

**Review by amymok — APPROVED** [review]
LGTM!

**Pull request merged**